app/src/androidTest/java/com/github/polypoly/app/base/game/GameTest.kt

```diff
@@ -146,4 +147,54 @@ class GameTest {
                 }
         }
     }
+
+    @Test
+    fun whenNextTurnComputesTransactions() {
+
+        gameLobby.start()
+        val game = Game.gameInProgress!!
+        val player1 = game.players[0]
+        val player2 = game.players[1]
+        val player1MoneyBefore = player1.getBalance()
+        val player2MoneyBefore = player2.getBalance()
+
+        class MockExecutedTransaction :
+            GameTransaction("Executed Mock Transaction", Player(), true) {
+
+            override fun executeTransaction() {
+                player1.earnMoney(100)
+            }
+
+            override fun getTransactionDescription(): String {
+                return "This is a mock transaction that has been executed :) !!!"
+            }
+        }
+
+        class MockNotExecutedTransaction :
+            GameTransaction("Not Executed Mock Transaction", Player(), false) {
+
+            override fun executeTransaction() {
+                player2.earnMoney(100)
+            }
+
+            override fun getTransactionDescription(): String {
+                return "This is a mock transaction that is not executed yet :) !!!"
+            }
+        }
+
+        game.transactions.add(MockExecutedTransaction())
+        game.transactions.add(MockNotExecutedTransaction())
+
+        game.nextTurn()
+
+        assert(game.transactions.isEmpty())
+        for (transactions in game.pastTransactions){
+            for (transaction in transactions) {
+                assertTrue(transaction.isExecuted())
+            }
+        }
+
+        assert(player1MoneyBefore + 100 != player1.getBalance())
```

Comment:
Shouldn't it be `==` instead of `!=`

Reply:
no, actually player1 balance is modified in the execution of the first mock transaction, that is supposed to be executed aready! So this player should in fact not have gained money after the transactions computing in game :)